source/includes/_endpoint-deck.md

```diff
@@ -0,0 +1,413 @@
+
+#### GET
+
+GET a deck entity returns a shoji:entity with all of its attributes:
+
+```json
+{
+    "element": "shoji:entity",
+    "self": "https://beta.crunch.io/api/datasets/223fd4/decks/223fd4/",
+    "body": {
+        "name": "Presentation deck",
+        "id": "223fd4",
+        "creation_time": "1987-10-15T11:45:00",
+        "description": "Explanation about the deck",
+        "is_public": false,
+        "owner_id": "https://beta.crunch.io/api/users/abcd3/",
+        "owner_name": "Real Person"
+    }
+}
+```
+
+Name | Type | Description
+---- | ---- | -----------
+name | string | Human-friendly string identifier
+id | string | Global unique identifier for this deck
+creation_time | timestamp | Time when this deck was created
+description | string | Longer annotations for this deck
+is_public | boolean | Indicates whether this is a public deck or not
+owner_id | url | Points to the owner of this deck
+owner_name | string | Name of the owner of the deck (referred by `owner_id`)
+
+#### PATCH
+
+To edit a deck, PATCH it with a
+shoji:entity. The server will return a 204 response on success or 400
+in case of a problem.
+
+```json
+{
+    "element": "shoji:entity",
+    "self": "https://beta.crunch.io/api/datasets/223fd4/decks/223fd4/",
+    "body": {
+        "name": "Presentation deck",
+        "id": "223fd4",
+        "creation_time": "1987-10-15T11:45:00",
+        "description": "Explanation about the deck"
+    }
+}
+```
+```http
+HTTP/1.1 204 No Content
+```
+
+For decks that the current user owns, "name", "description", and "is_public" are editable. Users with edit permissions on the dataset may edit "name" and "description" of public decks, but they cannot edit "is_public" unless they own the deck. Other deck attributes are not editable.
+
+### Order
+
+`/datasets/{id}/decks/order/`
+
+The deck order resource allows the user to arrange how API clients, such as the web application, will present the deck catalog. The deck order contains all decks that are visible to the current user, both personal and public.
+
+#### GET
+
+Returns a (Shoji Order)[#shoji-order] payload.
+
+```json
+{
+  "element": "shoji:order",
+  "self": "https://beta.crunch.io/api/datasets/223fd4/decks/order/",
+  "graph": [
+    "https://beta.crunch.io/api/datasets/223fd4/decks/1/",
+    "https://beta.crunch.io/api/datasets/223fd4/decks/2/",
+    "https://beta.crunch.io/api/datasets/223fd4/decks/3/"
+  ]
+}
+
+```
+
+#### PATCH
+
+PATCH the order resource to change the order of the decks. A 204 response indicates success.
+
+If the PATCH payload contains only a subset of available decks, those
+decks not referenced will be appended at the bottom of the top level graph in arbitrary
+order.
+
+```json
+{
+  "element": "shoji:order",
+  "self": "https://beta.crunch.io/api/datasets/223fd4/decks/order/",
+  "graph": [
+    "https://beta.crunch.io/api/datasets/223fd4/decks/1/",
+    "https://beta.crunch.io/api/datasets/223fd4/decks/3/"
+  ]
+}
+```
+
```

Comment:
We do PUT=PATCH in the code so both handlers do the same. We're implementing the PATCH most of the times and then just assigning. The semantics for partial orders was never defined so we had them both do the same.